lib/src/utils/gtk_colors.dart

```diff
@@ -1,6 +1,85 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/foundation.dart';
 
+import 'dart:io';
+import 'dart:ui';
+import 'package:from_css_color/from_css_color.dart';
+import 'package:gsettings/gsettings.dart';
+
+class GnomeTheme {
+  String contents = "";
+
+  Color? themeBgColor;
+  Color? textColor;
+  Color? buttonFg;
+  Color? themeBaseColor;
+  Color? insensitiveBgColor;
+  // Border of widgets not text inputs fields
+  Color? unfocusedBorder;
+  Color? border;
+  Color? themeSelectedBG;
+  Color? textSelectedFG;
+
+  GnomeTheme() {
+    loadFromFile();
+    parse();
+  }
+
+  void loadFromFile() {
+    final settings = GSettings(schemaId: 'org.gnome.desktop.interface');
+    final theme = settings.stringValue('gtk-theme');
+    String fileName = "/usr/share/themes/" + theme + "/gtk-3.0/gtk.css";
```

Comment:
The file might also be at ~/.themes or ~/.local/share/themes, as their might be some user defined themes.

You can either use xdg_directories plugin or a manual way using path_provider package.